game/state/gamestate.cpp

```diff
@@ -1178,7 +1192,141 @@ void GameState::updateEndOfDay()
 	}
 }
 
-void GameState::updateEndOfWeek() { luaGameState.callHook("updateEndOfWeek", 0, 0); }
+void GameState::updateEndOfWeek()
+{
+	updateHumanEconomy();
+
+	luaGameState.callHook("updateEndOfWeek", 0, 0);
+}
+
+// Recalculates AI organization and civilian finances, updating budgets and salaries
+void GameState::updateHumanEconomy()
+{
+	// TODO: remove hardcoded references
+	auto humanCity = cities["CITYMAP_HUMAN"];
+
+	humanCity->populationWorking = 0;
+	// Game resets only Government income, it's not right logically but will keep it to match OG
+	government->income = 0;
+
+	// Step 1. Everybody gets paid according to the current rates
+	int totalCivilianIncome = 0;
+	for (auto &org : organisations)
+	{
+		if (org.first != player.id && org.first != aliens.id)
+		{
+			for (auto &b : org.second->buildings)
+			{
+				// validate the original data
+				if (b->currentWage < 0)
+					b->currentWage = 0;
+
+				org.second->income += b->calculateIncome();
+				humanCity->populationWorking += b->currentWorkforce;
+				totalCivilianIncome += b->currentWage * b->currentWorkforce;
+			}
+			org.second->balance += org.second->income;
+		}
+	}
+	humanCity->averageWage =
+	    (humanCity->populationWorking) ? totalCivilianIncome / humanCity->populationWorking : 0;
+
+	// Step 2. Government additionally gets 10% of civilian income as taxes
+	government->balance += totalCivilianIncome / 10;
+
+	// Step 3. Calculate civilians leaving work because of the low wage
+	const int minimumWage = std::max(humanCity->averageWage, 30);
+	for (auto &pair : humanCity->buildings)
```

Comment:
Can you use a structured binding? (IE const auto& [id, build] : humanCity->buildings)?